Tests/Shared.Specs/StringAssertionSpecs.cs

```diff
@@ -1785,43 +1785,904 @@ public void When_asserting_string_does_not_end_with_equivalent_of_a_value_and_ac
+            act.Should().Throw<XunitException>().WithMessage(
+                "Expected string \"ABCDEF\" to contain \"XYZ\" because that is required.");
+        }
+
+        [Fact]
+        public void When_containment_is_asserted_against_null_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => "a".Should().Contain(null);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act
+                .Should().Throw<ArgumentNullException>()
+                .WithMessage("Cannot assert string containment against <null>.*");
+        }
+
+        [Fact]
+        public void When_containment_is_asserted_against_an_empty_string_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => "a".Should().Contain("");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act
+                .Should().Throw<ArgumentException>()
+                .WithMessage("Cannot assert string containment against an empty string.*");
+        }
+
+        [Fact]
+        public void When_string_containment_is_asserted_and_actual_value_is_null_then_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            string someString = null;
+            Action act = () => someString.Should().Contain("XYZ", "that is {0}", "required");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().Throw<XunitException>().WithMessage(
+#if NETCOREAPP1_1
+                 "Expected string <null> to contain \"XYZ\" because that is required.");
```

Comment:
🙃 As we're not testing the context replacement, I would remove the preprocessor directive and replace `string`/`someString` with `*`.